networking/src/p2p/peer.rs

```diff
@@ -217,80 +214,49 @@ impl Actor for Peer {
     }
 }
 
-impl Receive<Bootstrap> for Peer {
+impl Receive<BootstrapOutput> for Peer {
     type Msg = PeerMsg;
 
-    fn receive(&mut self, ctx: &Context<Self::Msg>, msg: Bootstrap, _sender: Sender) {
-        let info = self.local.clone();
+    fn receive(&mut self, ctx: &Context<Self::Msg>, msg: BootstrapOutput, _sender: Sender) {
+        let BootstrapOutput(rx, tx, peer_public_key_hash, peer_id_marker, peer_metadata, peer_compatible_network_version, peer_address) = msg;
+
         let myself = ctx.myself();
         let system = ctx.system.clone();
-        let net = self.net.clone();
+        let mut net = self.net.clone();
         let network_channel = self.network_channel.clone();
-        self.remote_addr = msg.address;
+        self.remote_addr = peer_address;
 
         self.tokio_executor.spawn(async move {
```

Comment:
@vlad9486 
ok, very cool,
and last thing I would like to move all this stuff `self.tokio_executor.spawn(async move {` to `fn create_args`,
and remove `impl Receive<BootstrapOutput>`, so BootstrapOutput will be an argument of `fn actor(`

so in peer_manager, there will be just `Self::create_peer(` and no `peer.tell(bootstrap_output, None)`,
so this will be the final: peer_manager just calls `create_peer`